crates/holochain_zome_types/src/warrant.rs

```diff
@@ -67,6 +67,7 @@ impl HashableContent for Warrant {
 #[derive(
     Clone, Debug, Serialize, Deserialize, SerializedBytes, Eq, PartialEq, Hash, derive_more::From,
 )]
+#[serde(tag = "type", content = "value", rename_all = "snake_case")]
 pub enum WarrantProof {
```

Comment:
Suggest that app backends should contain most of the logic for dealing with warrants. To me they're in the same category as `Record`s and should be handled by the coordinator. If an app chooses to return them then I think it's reasonable to say that they should receive them in the default format that Holochain has them in

Reply:
They are returned by the state dump and may therefore be used in happ frontends for debuggability purposes. Moss is doing this for example, although it's not using a holochain version that would actually ever produce warrants I think.

Reply:
Understood that it's used in dumps. I think that's then an API mistake to return internal types that need to be serialized a certain way. The dump should be translating to appropriate types for clients

Reply:
> I think that's then an API mistake to return internal types that need to be serialized a certain way

In what way is this a mistake? Is it causing any problems somewhere? I would naively expect serialization not to matter at all on the Rust side when it comes to internal use of these types...

Assuming there *is* an actual problem, are you suggesting that this PR should now also include logic to convert the state dump results into dedicated types to return to the clients?